cookbook/bundles/tests.rst

```diff
@@ -0,0 +1,311 @@
+
+.. code-block:: bash
+
+    $ php Tests/app/console.php server:run -d Tests/app
+
+.. note::
+
+    If the third party bundle uses the Twig templating engine to render HTML pages
+    or if it uses the Form Component or anything else, then more dependencies and
+    configuration parameters should be added.
+
+Running Commands
+----------------
+
+If the third party bundle provides a command, it can be helpful to create a ``console``
+similar to the one that can be found in Symfony's Standard Edition::
+
+    // Tests/app/console.php
+    set_time_limit(0);
+
+    require_once __DIR__.'/autoload.php';
+
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+    $kernel = new AppKernel('dev', true);
+    $application = new Application($kernel);
+    $application->run();
+
+With this it becomes possible to run manually the console:
+
+.. code-block:: bash
+
+    $ php Tests/app/console.php
+
+Automated Tests
+---------------
+
+Manual tests are great to get a quick idea of what the bundle does. But writing
+automated tests is even better!
+
+The first step is to install a test framework like PHPUnit:
+
+.. code-block:: bash
+
+    $ composer require --dev phpunit/phpunit
+
+.. note::
+
+    The steps should be similar with other tests frameworks.
+
+Then the second one is to configure it to use Composer's autoloading:
+
+.. configuration-block::
+
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8"?>
+        <!-- phpunit.xml.dist -->
+
+        <!-- http://phpunit.de/manual/4.3/en/appendixes.configuration.html -->
+        <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.3/phpunit.xsd"
+            backupGlobals="false"
+            colors="true"
+            bootstrap="./vendor/autoload.php"
+        >
+            <testsuites>
+                <testsuite name="Test Suite">
+                    <directory>./Tests/</directory>
+                </testsuite>
+            </testsuites>
+
+            <filter>
+                <whitelist>
+                    <directory>.</directory>
+                    <exclude>
+                        <directory>./Resources</directory>
+                        <directory>./Tests</directory>
+                    </exclude>
+                </whitelist>
+            </filter>
+        </phpunit>
+
+With these two simple steps it becomes possible to run the test suite with the following command:
+
+.. code-block:: bash
+
+    $ vendor/bin/phpunit
+
+Functional Web Tests
+~~~~~~~~~~~~~~~~~~~~
+
+As advised in the official best practices (smoke testing), writing tests for
+controllers can be done by simply checking the status code::
+
+    // Tests/Controller/DemoControllerTest.php
+    namespace Acme\MyBundle\Tests\Controller;
+
+    use Acme\MyBundle\Tests\AppKernel;
+    use Symfony\Component\HttpFoundation\Request;
+
+    class DemoControllerTest extends \PHPUnit_Framework_TestCase
```

Comment:
There's no point in using `WebTestCase` here.
Also testing the content defeats the point of the article (which is promoting smoke testing as suggested in the official best practices).
This approach is driven by the fact that the content will change often making them fragile and the value provided by those checks don't make it worth it.